test/squish/suite_VerifySettingsDialog/tst_VerifySettingsDialog/test.py

```diff
@@ -0,0 +1,28 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# mypy: disable-error-code="attr-defined"
+
+import gui_helpers
+import test
+
+profileName = "deadlinecloud_squish"
+jobHistDir = "~/.deadline/job_history/deadlinecloud_squish"
+farmName = "Deadline Cloud Squish Farm"
+queueName = "Squish Automation Queue"
+jobAttachments = "COPIED"
+storageProfile = "Squish Storage Profile"
+conflictResOption = "NOT\\_SELECTED"
+loggingLevel = "WARNING"
+
+
+def main():
+    gui_helpers.launchDeadlineConfigGUI()
+    gui_helpers.setGlobalSettings(profileName)
+    gui_helpers.setProfileSettings(jobHistDir, farmName)
+    gui_helpers.openAndSetDefaultJobHistDirectory()
+    gui_helpers.setFarmSettings(queueName, storageProfile, jobAttachments)
+    gui_helpers.setGeneralSettings(conflictResOption, loggingLevel)
+    # disabling due to https://sim.amazon.com/issues/Bea-28289
+    # gui_helpers.hitApplyButton()
+    test.log("All settings config have been applied.")
+    gui_helpers.closeDeadlineConfigGUI()
```

Comment:
If one of the tests above fail, does this line still get called? Is the machine that was running the tests left in a weird state?

Reply:
No - Currently, if a test fails, Squish will shut down the GUI (and any new tests would pretty much be started up again and any failed states won't affect it). But I do agree we should add 'beforeAll' or 'afterAll' blocks to set the test up and help clean things up.

Reply:
It likely won't be a Python test setup thing but something specific to Squish. I'm pretty sure they have the concept of test suites that you can setup and have standard setup and teardown. 

If these tests are run as is on a developers laptop, will they leave state behind say the next time the just run the deadline CLI on their own?

Reply:
Yes - It looks like Squish does have some concepts on setup/teardown. Reference is here: https://doc.qt.io/squish/tester-created-special-functions.html

I was thinking about this, and I can't think of any setup/teardown function we should add other than login and logout functions (for the dialogue). The farm resources that appear in the dialogue pretty much depend on what AWS profile you are logged into, so the best thing to do would be to logout of the dialogue when the test finishes (and then the developer would need to log back in themselves and just re-select the profile they want to use). Additionally, when the test re-runs, the test would need to log back in as well (Once logged back in, the test would proceed and select the correct AWS profile, farm resources, verify those resources, etc). 

With that being said, I tried importing the Deadline CLI API into the Squish framework and tested out the login/logout api/functions from the Deadline API. The login api/function had some issues. There were errors saying that the AWS Profile had to be created by DCM. Additionally, I also had to modify some code in the login api/function so to get it working when importing the api so it makes me think there is a bug with the login api/function since it didn't work right off the bat. But the error saying that the 'AWS Profile had to be created by DCM' makes me think getting the login api/function to work won't be that simple. 

I tried the logout api/function and it worked right off the bat and was usable. 

I'll need to look into this further once I am back from leave.